File: RetrofitExample/app/src/main/java/com/example/velm/retrofitexample/MainActivity.java
package com.example.velm.retrofitexample;

import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;

import java.util.ArrayList;
import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {


    private List<Movie> movieList;
    private RecyclerView recyclerView;

    private RecyclerAdapter recyclerAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        movieList = new ArrayList<>();

        recyclerView = (RecyclerView)findViewById(R.id.recyclerview);


        LinearLayoutManager layoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(layoutManager);


        ApiInterface apiService = ApiClient.getClient().create(ApiInterface.class);

        Call<List<Movie>> call = apiService.getMovies();

        call.enqueue(new Callback<List<Movie>>() {
            @Override
            public void onResponse(@NonNull Call<List<Movie>> call, @NonNull Response<List<Movie>> response) {

                movieList = response.body();
                Log.d("TAG","Response = "+movieList);
                recyclerAdapter = new RecyclerAdapter(getApplicationContext(),movieList);
                recyclerView.setAdapter(recyclerAdapter);

            }

            @Override
            public void onFailure(@NonNull Call<List<Movie>> call2, @NonNull Throwable t) {
                Log.d("TAG","Failed Response = "+t.toString());

                //call2.cancel();
                Log.d("TAG","canceled");
            }
        });

    }

    /*@Override
    protected void onDestroy() {
        super.onDestroy();
        Log.d("TAG","Ondestroy");
        if(call.isExecuted()){
            call.cancel();
            Log.d("TAG","call canedlled");
        } else {
            Log.d("TAG","NO calls");
        }
    }*/
}

File: dragger2/src/main/java/app/com/dragger2/MainActivity.kt
package app.com.dragger2

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit
import javax.inject.Inject

class MainActivity : AppCompatActivity() {

    @Inject lateinit var retrofit: Retrofit

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        DaggerApiComponent.builder()
                .appModule(AppModule(MyApp().get(this)))
                .apiModule(ApiModule())
                .build()
                .inject(this)


        val all = retrofit.create(Api::class.java).getAll()


        all.enqueue(object : Callback<List<User>>{
            override fun onFailure(call: Call<List<User>>, t: Throwable) {
                Log.d("Failed = ", t.toString())
            }

            override fun onResponse(call: Call<List<User>>, response: Response<List<User>>) {
                Log.d("Response = ", response.body().toString())
            }
        })
    }


}


